src/pixel/mod.rs

```diff
@@ -10,6 +10,15 @@ pub use self::gamma_rgb::GammaRgb;
 mod srgb;
 mod gamma_rgb;
 
+///A transfer function to and from linear space.
+pub trait TransferFn {
```

Comment:
To the TransferFn. Hsl to LinearHsl, should convert though rgb. So that is the only conversion TransferFn should care about.